src/main/io/dataEdition.c

```diff
@@ -0,0 +1,1359 @@
+	        		getReceiverMapping();
+					structureOfPage.editField[selectedItem].color = PIXEL_MIXED;
+	        		structureOfPage.editField[selectedItem].data = getChannelIndex();
+				}
+#endif
+			}
+		}
+		else
+			exitEditionProcess();
+	}
+	else
+
+	switch (commandID) {
+		case INC_COMMAND : {
+			processIncDecEvent(1);
+			break;
+		}
+		case DEC_COMMAND : {
+			processIncDecEvent(-1);
+			break;
+		}
+		case NEXT_COMMAND : {
+			processNextPrevEvent (1);
+			break;
+		}
+		case PREV_COMMAND : {
+			processNextPrevEvent (-1);
+			break;
+		}
+		case SET_COMMAND : {
+#if (DATA_EDITION_FULL_VERSION)
+			if ((selectedItem == 21) && (selectedPage != MOTORS_PAGE)){
+			    saveConfigAndNotify();
+			    structureOfPage.editField[selectedItem].color = PIXEL_NORMAL;
+				exitEditionProcess();
+			}
+			else {
+				if ((selectedPage == RECEIVER_PAGE) && (selectedItem == 18)){
+					if (structureOfPage.editField[selectedItem].color == PIXEL_INVERTED)
+						structureOfPage.editField[selectedItem].color = PIXEL_MIXED;
+					else {
+						if (theRCMappingStringIsValid()){
+							parseRcChannels(tempString, &masterConfig.rxConfig);
+							structureOfPage.editField[selectedItem].color = PIXEL_NORMAL;
+							exitEditionProcess();
+						}
+						else {
+							structureOfPage.editField[selectedItem].color = PIXEL_INVERTED;
+					        beeperConfirmationBeeps(2);
+						}
+					}
+				}
+				else
+					if (onGoingEditionProcess){
+						structureOfPage.editField[selectedItem].color = PIXEL_NORMAL;
+						exitEditionProcess();
+					}
+			}
+#else
+			if (selectedItem == 21){
+				saveConfigAndNotify();
+				structureOfPage.editField[selectedItem].color = PIXEL_NORMAL;
+				exitEditionProcess();
+			}
+			else
+				if (onGoingEditionProcess){
+					structureOfPage.editField[selectedItem].color = PIXEL_NORMAL;
+					exitEditionProcess();
+				}
+#endif
+			break;
+		}
+	}
+}
+
+//=======================  end of processing interface of the page  ========================
+
+//=================================  process SelectedPage  ==================================
+static void writeSignOfSelectedLine(void)
+{
+	contentOfPage.lineOfText[selectedLine][0] = 0x3E;
+}
+
+static void processSelectedPage(uint8_t commandID)
+{
+    if (shouldInitialiseThePage)
+    	initializeNewPage();
+
+	processInterfaceOfThePage(commandID);
+ 	updateValuesForThatPage();
+ 	prepareLinesOfTextForThatPage();
+
+#if (DATA_EDITION_FULL_VERSION)
+    if (selectedPage != FIRMWARE_PAGE)
+    	writeSignOfSelectedLine();
+#else
+	writeSignOfSelectedLine();
+#endif
+}
+
+//==============================  processing data Edition  =================================
+void sendContentOfPageToDisplay(void)
+{
+	char tmpChar[22];
+
+	if ((featureConfigured(FEATURE_DISPLAY)) && (onGoingDataEditionWithSticks)){
+		for (int i = 0; i <= 7; i++) {
+			snprintf(tmpChar,DEFAULT_DISPLAY_COLUMNS+1, contentOfPage.lineOfText[i]);
+			tmpChar[DEFAULT_DISPLAY_COLUMNS] = 0;
```

Comment:
why `snprintf` with `DEFAULT_DISPLAY_COLUMNS+1` and then truncate last character?

also snprintf always terminates output with '\0'